File: Coding/Java/mergeSort.java
import java.util.Scanner;
// Other imports go here
// Do NOT change the class name
class mergeSort
{
    public static void mergeSort(int a[])
    {
        if(a.length>1)
        {
        int mid=a.length/2;
        int i,j;
        int left[]=new int[mid];
        int right[]=new int[a.length-mid];
        for(i=0;i<mid;i++)
        {
            left[i]=a[i];
        }
        for(i=mid,j=0;i<a.length;i++,j++)
        {
            right[j]=a[i];
        }
        mergeSort(left);
        mergeSort(right);
        merge(left,right,a);
        }
        else
        return;
    }
    public static void merge(int left[],int right[],int a[])
    {
        int i=0,j=0,k=0;
        int l1=left.length;
        int r1=right.length;
        int a1=a.length;
        while(i<l1 && j<r1)
        {
          if(left[i]<right[j])
          {
              a[k]=left[i];
              k++;
              i++;
          }
          else
          {
              a[k]=right[j];
              k++;
              j++;
          }    
        }
        while(i<l1)
        {
            a[k]=left[i];
            k++;
            i++;
        }
        while(j<r1)
        {
            a[k]=right[j];
            j++;
            k++;
        }
    }
    public static void main(String[] args)
    {
        // Write your code here
        Scanner s=new Scanner(System.in);
System.out.print("Enter number of elements");
        int n=s.nextInt();
        int arr[]=new int[n];
        for(int i=0;i<n;i++)
        {
            arr[i]=s.nextInt();
        }
        mergeSort(arr);
        for(int i=0;i<n;i++)
        {
            System.out.print(arr[i]+" ");
        }
    }
    }

File: Coding/C/MergeSort.c
#include <stdio.h>
void merge(int *ar, int left,int right, int mid)
{
    int n1,n2,i,j,k;
    n1=mid-left+2;
    n2=right-mid+1;
    int l[n1],r[n2];
    for(i=0;i<n1-1;i++)
    {
        l[i]=ar[left+i];
    }
    for(i=0;i<n2-1;i++)
    {
        r[i]=ar[mid+1+i];
    }
    l[n1-1]=9999;
    r[n2-1]=9999;
    i=0;
    j=0;
    for(k=left;k<=right;k++)
    {
        if(l[i]<r[j])
        {
            ar[k]=l[i];
            i++;
        }
        else
        {
            ar[k]=r[j];
            j++;
        }
    }
}
void mergesort(int *ar,int left,int right)
{
    int i,j,mid;
    if(left<right)
    {
        mid=(left+right)/2;
        mergesort(ar,left,mid);
        mergesort(ar,mid+1,right);
        merge(ar,left,right,mid);
    }
}
int main()
{
int n,i,j;
scanf("%d",&n);
int ar[n];
for(i=0;i<n;i++)
{
    scanf("%d",&ar[i]);
}
mergesort(ar,0,n-1);
for(i=0;i<n;i++)
{
    printf("%d ",ar[i]);
}
    return 0;
}

